crates/update-notifier/src/ui.rs

```diff
@@ -0,0 +1,66 @@
+use terminal_size::{terminal_size, Width};
+
+use crate::UpdateNotifierError;
+pub mod utils;
+
+const DEFAULT_PADDING: usize = 8;
+
+pub fn message(text: &str) -> Result<(), UpdateNotifierError> {
+    let size = terminal_size();
+    let lines: Vec<&str> = text.split('\n').map(|line| line.trim()).collect();
+
+    // get the display width of each line so we can center it within the box later
+    let lines_display_width: Result<Vec<usize>, utils::GetDisplayLengthError> = lines
+        .iter()
+        .map(|line| utils::get_display_length(line))
+        .collect();
+
+    let lines_display_width = match lines_display_width {
```

Comment:
Pretty sure you could use a `?` here instead. In fact you could probably do: 
```
  let lines_display_width = lines
         .iter()
         .map(|line| utils::get_display_length(line))
         .collect::<Result<Vec<_>, _>>()?;
```

But that's a bit of a stylistic decision.

Reply:
Nice!